### PR TITLE
Move capi obj key method to controller capi file

### PR DESCRIPTION
**Pull request by taneyland**
*Issue #, if available:*

*Description of changes:*
Follow up from this [discussion](https://github.com/aws/eks-anywhere/pull/3192#discussion_r957774271)

*Testing (if applicable):*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: pkg/controller/clusters/clusterapi.go [hunk 1]
```diff
@@ -37,3 +38,11 @@ func CheckControlPlaneReady(ctx context.Context, client client.Client, log logr.
 	log.Info("CAPI control plane is ready")
 	return controller.Result{}, nil
 }
+
+func CapiClusterObjectKey(cluster *anywherev1.Cluster) client.ObjectKey {
```
Comment (g-gaston) [inline pkg/controller/clusters/clusterapi.go:42]: nit: add doc comment?
```suggestion
// CapiClusterObjectKey generates and ObjectKey for the CAPI cluster owned by
// the provided eks-a cluster
func CapiClusterObjectKey(cluster *anywherev1.Cluster) client.ObjectKey {
```

File: pkg/controller/clusters/clusterapi_test.go [hunk 2]
```diff
@@ -94,6 +96,19 @@ func eksaCluster() *anywherev1.Cluster {
...
 }
 
+func TestGetCapiClusterObjectKey(t *testing.T) {
+	g := NewWithT(t)
+	eksaCluster := eksaCluster()
+
+	expected := types.NamespacedName{
+		Name:      clusterapi.ClusterName(eksaCluster),
+		Namespace: constants.EksaSystemNamespace,
+	}
```
Comment (g-gaston) [inline pkg/controller/clusters/clusterapi_test.go:106]: IMO we shouldn't use the same code being tested to generate the expected results

```suggestion
	expected := types.NamespacedName{
		Name:      "my-cluster",
		Namespace: "eksa-system",
	}
```

File: pkg/controller/clusters/clusterapi.go (hunk 1 above)
Comment (g-gaston) [inline pkg/controller/clusters/clusterapi.go:42]: I would maybe move this to `pkg/controller/clusterapi.go`
It seems something closer to what we have there, I think.

Comment (taneyland) [reply pkg/controller/clusters/clusterapi.go]: Makes sense. I was on the fence on which file to use so I definitely see your point, updated it!

File: pkg/controller/clusterapi.go [hunk 3]
```diff
@@ -31,3 +31,13 @@ func GetCAPICluster(ctx context.Context, client client.Client, cluster *anywhere
 	}
 	return capiCluster, nil
 }
+
+// CapiClusterObjectKey generates an ObjectKey for the CAPI cluster owned by
+// the provided eks-a cluster
+func CapiClusterObjectKey(cluster *anywherev1.Cluster) client.ObjectKey {
```
Comment (g-gaston) [inline pkg/controller/clusterapi.go:37]: sorry i missed this
```suggestion
// CAPIClusterObjectKey generates an ObjectKey for the CAPI cluster owned by
// the provided eks-a cluster
func CAPIClusterObjectKey(cluster *anywherev1.Cluster) client.ObjectKey {
```

**Review by g-gaston — APPROVED** [review]

**Pull request merged**